core/shared/src/main/scala/fs2/compression/GunzipResult.scala

```diff
@@ -0,0 +1,159 @@
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2
+package compression
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.FiniteDuration
+
+/** Gunzip decompression results including file properties and
+  * decompressed content stream, used as follows:
+  *   stream
+  *     .through(gunzip[IO]())
+  *     .flatMap { gunzipResult =>
+  *       // Access properties here.
+  *       gunzipResult.content
+  *     }
+  *
+  * @param content Uncompressed content stream.
+  * @param modificationTime Modification time of compressed file.
+  * @param fileName File name.
+  * @param comment File comment.
+  */
+class GunzipResult[F[_]](
+    val modificationTimeEpoch: Option[FiniteDuration],
+    val fileName: Option[String],
+    val comment: Option[String],
+    val content: Stream[F, Byte]
+) extends Product
+    with Serializable
+    with Equals {
+
+  def modificationTime: Option[Instant] =
+    modificationTimeEpoch.map(e => Instant.ofEpochSecond(e.toSeconds))
+
+  @deprecated(
+    "use the default constructor with modificationTimeEpoch: Option[FiniteDuration]",
+    "x.x"
+  )
+  def this(
+      content: Stream[F, Byte],
+      modificationTime: Option[Instant] = None,
+      fileName: Option[String] = None,
+      comment: Option[String] = None
+  ) = this(
+    modificationTime.map(i => FiniteDuration(i.getEpochSecond, TimeUnit.SECONDS)),
+    fileName,
+    comment,
+    content
+  )
+
+  @deprecated("GunzipResult is no longer a case class.", "x.x")
+  def copy(
+      content: Stream[F, Byte] = this.content,
+      modificationTime: Option[Instant] = this.modificationTime,
+      fileName: Option[String] = this.fileName,
+      comment: Option[String] = this.comment
+  ): GunzipResult[F] = new GunzipResult[F](
+    content,
+    modificationTime,
+    fileName,
+    comment
+  )
+
+  @deprecated("GunzipResult is no longer a case class.", "x.x")
+  def productArity: Int = 4
+
+  @deprecated("GunzipResult is no longer a case class.", "x.x")
+  def productElement(n: Int): Any = n match {
+    case 0 => content
+    case 1 => modificationTimeEpoch
```

Comment:
```suggestion
    case 1 => modificationTime
```

Reply:
`ioJS/ testOnly fs2.io.NodeJSCompressionSuite`:

```
[error] Referring to non-existent class java.time.Instant
[error]   called from private fs2.compression.GunzipResult.$anonfun$modificationTime$1(scala.concurrent.duration.FiniteDuration)java.time.Instant
```

🤔

Reply:
Crap. Also, I bet we have this same problem in http4s with JVM `Path`, except because of the munit bug our test suite doesn't catch it.

One work-around is to put this into platform-specific trait. But also probably not worth the trouble, I am not sure how/why this method would be used.

Reply:
Ah yeah, I still remember that one 🤔

Reply:
I was thinking which one the `productElement` should return, and then forgot to think about it more.